lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -296,7 +416,7 @@ def wrap_init(self: CharmBase, framework: Framework, *args, **kwargs):
 
         original_event_context = framework._event_context
 
-        _service_name = service_name or self.app.name
+        _service_name = service_name or self.unit.name.replace("/", "-")
```

Comment:
Not certain if that move is in the right direction. It means that in some scenarios each of the units is seen as a separate service in tempo, right? I'd rather add `unit` as one of the tags of created spans as in case of fast-changing units this could quickly increase the number of services that are seen in tempo search on Grafana to the point of it not being readable.

Reply:
mmm, that's only going to be a problem if:
- we have TONS of units
- we are on a machine model, where unit IDs are not recycled if you respawn them.

I see the danger, however, I think there is no point in doing charm tracing if we don't know exactly which charm trace we are looking at.

If you have many units and you do a blanket search and you get too many results, you should IMHO simply narrow down your query to a range of unit IDs you're interested in

Reply:
or are you suggesting to split service and unit id to two separate labels?

Reply:
Yes, that was my suggestion - to split service name from the unit ID (which should stay as a label, we want to be able to filter by a specific unit) - unless we want to keep charm traces separate from the workload traces. (btw now that I think about it I wonder if charm traces shouldn't have a label distinguishing them from other traces)

In most cases I don't expect we (or rather the charm developer) are passing unit id as part of the service name to workloads so that they send traces with the right context.  However I might be wrong :)